cloudferrylib/os/image/glance_image.py

```diff
@@ -227,8 +227,8 @@ def delete_image(self, image_id):
         self.glance_client.images.update(image_id, protected=False)
         self.glance_client.images.delete(image_id)
 
-    def get_image_by_id(self, image_id):
-        for glance_image in self.get_image_list():
+    def get_image_by_id(self, image_id, is_filtering=True):
+        for glance_image in self.get_image_list(is_filtering):
```

Comment:
Please use `glance_client.get`

    try:
        self.glance_client.images.get(image_id)
    except NotFound:
        LOG.warning('Image %s not found on %s', image_id, self.cloud.position)
        return None

This function code was bad before your commit: getting list of images from API every time it is called is not good for performance. And now it become even worse. I think we should improve code whenever we can, and you have good chance to do it here.

As for possible problems with changing this code: this function used in 14 places, it's not hard to check them all.